provider/azure/instance.go

```diff
@@ -414,6 +416,9 @@ func deleteInstanceNetworkSecurityRules(
 ) error {
 	nsg, err := nsgClient.Get(resourceGroup, internalSecurityGroupName, "")
 	if err != nil {
+		if err.(autorest.DetailedError).Response.StatusCode == http.StatusNotFound {
```

Comment:
I'd suggest being defensive here, because if for some reason nsgClient.Get returns an error that isn't an instance of autorest.DetailedError, we'll panic here.

    if err, ok := err.(autorest.DetailedError); ok && err.Response.StatusCode == http.StatusNotFound {
        return nil
    }

Reply:
updated via pr8070